scrapy/core/http2/protocol.py

```diff
@@ -33,13 +33,17 @@
 logger = logging.getLogger(__name__)
 
 
+ACCEPTABLE_PROTOCOL = b"h2"
+
+
 class InvalidNegotiatedProtocol(H2Error):
 
-    def __init__(self, negotiated_protocol: str) -> None:
+    def __init__(self, negotiated_protocol: bytes) -> None:
         self.negotiated_protocol = negotiated_protocol
 
     def __str__(self) -> str:
-        return f'InvalidNegotiatedProtocol: Expected h2 as negotiated protocol, received {self.negotiated_protocol!r}'
+        return (f"Expected {ACCEPTABLE_PROTOCOL.decode('utf-8')} as negotiated protocol,"
+                f" received {self.negotiated_protocol.decode('utf-8')}")
```

Comment:
:lipstick: Instead of decoding, maybe we could use `!r` and let it be shown as `b'…'`. Otherwise, this will break on non-UTF-8 stuff, which while unlikely is still possible.

Reply:
Sure, makes sense